### PR TITLE
Update dataset schema limit

### PR DESCRIPTION
**Pull request by serena-ruan**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/serena-ruan/mlflow/pull/12603?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12603/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12603
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?
Update dataset schema limit to 16MB

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/utils/validation.py [hunk 1]
```diff
@@ -60,7 +60,7 @@
 MAX_EXPERIMENTS_LISTED_PER_PAGE = 50000
 MAX_DATASET_NAME_SIZE = 500
 MAX_DATASET_DIGEST_SIZE = 36
-MAX_DATASET_SCHEMA_SIZE = 65535  # 64KB -1 (the db limit for TEXT column)
+MAX_DATASET_SCHEMA_SIZE = 16777215  # 16MB -1 (the db limit for MEDIUMTEXT column)
```
Comment (WeichenXu123) [inline mlflow/utils/validation.py:63]: Q: Do we have so long schema string ?

Comment (serena-ruan) [reply mlflow/utils/validation.py:63]: Yes, databricks backend has supported it https://src.dev.databricks.com/databricks/universe@7860d65beb6559a9f7c4b4b68476174cab175ace/-/blob/mlflow/db/tracking/migrations/V46__datasets_change_schema_to_medium_text.jsonnet

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 2]
```diff
@@ -3431,7 +3433,7 @@ def test_log_inputs_fails_with_too_large_inputs(store: SqlAlchemyStore):
                         digest="digest",
                         source_type="type",
                         source="source",
-                        schema="a" * 65536,
+                        schema="a" * 16777216,
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:3436]:
```suggestion
                        schema="a" * MAX_DATASET_SCHEMA_SIZE + 1,
```

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 3]
```diff
@@ -3419,8 +3419,10 @@ def test_log_inputs_fails_with_too_large_inputs(store: SqlAlchemyStore):
             ],
         )
 
-    # Test dataset schema too large (limit is 65535)
-    with pytest.raises(MlflowException, match="Dataset schema exceeds the maximum length of 65535"):
+    # Test dataset schema too large (limit is 16777215)
+    with pytest.raises(
+        MlflowException, match="Dataset schema exceeds the maximum length of 16777215"
+    ):
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:3425]:
```suggestion
    # Test dataset schema too large
    with pytest.raises(
        MlflowException, match=r"Dataset schema exceeds the maximum length of \d+"
    ):
```

**Review by daniellok-db — APPROVED** [review]
lgtm

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 4]
```diff
@@ -3347,115 +3356,212 @@ def test_log_inputs_fails_with_missing_inputs(store: SqlAlchemyStore):
...
+    error_message,
+):
+    run = _run_factory(store, _get_run_configs(_create_experiments(store, exp_name)))
+    store.log_inputs(run.info.run_id, correct_dataset_inputs)
+    run1 = store.get_run(run.info.run_id)
+    assert_dataset_inputs_equal(run1.inputs.dataset_inputs, correct_dataset_inputs)
 
-    dataset = entities.Dataset(name="name1", digest="digest1", source_type="type", source="source")
+    with pytest.raises(MlflowException, match=error_message):
+        store.log_inputs(run.info.run_id, wrong_dataset_inputs)
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:3372]: I'd isolate the error check for readability.

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**